### ISSUE TITLE
**TypeError: item.level.toUpperCase() is not a function**

### ISSUE THREAD

**Issue by Juanelorganelo** [timdp/winston-aws-cloudwatch#80]
I'm getting an error using this transport, it appears to be from the library itself
Here is the stack trace
```
TypeError: item.level.toUpperCase is not a function
    at CloudWatchEventFormatter.formatLog (/home/user/documents/library/node_modules/winston-aws-cloudwatch/lib/cloudwatch-event-formatter.js:35:27)
    at CloudWatchEventFormatter.formatLogItem (/home/user/documents/library/node_modules/winston-aws-cloudwatch/lib/cloudwatch-event-formatter.js:28:21)
    at params.logEvents.batch.map.item (/home/user/documents/library/node_modules/winston-aws-cloudwatch/lib/cloudwatch-client.js:120:52)
    at Array.map (<anonymous>)
    at CloudWatchClient._putLogEvents (/home/user/documents/library/node_modules/winston-aws-cloudwatch/lib/cloudwatch-client.js:120:24)
    at CloudWatchClient._putLogEventsAndStoreSequenceToken (/home/user/documents/library/node_modules/winston-aws-cloudwatch/lib/cloudwatch-client.js:111:17)
    at _maybeUpdateSequenceToken.then (/home/user/documents/library/node_modules/winston-aws-cloudwatch/lib/cloudwatch-client.js:90:61)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:160:7)
```

here is the code

```js
export const log = winston.createLogger({
  level: 'debug',
  format: winston.format.json(),
  transports: [
    new CloudWatchTransport({
      level: 'error',
      createLogGroup: true,
      createLogStream: true,
      logGroupName: AWS_CLOUDWATCH_GROUP,
      logStreamName: 'errors',
    }),
  ],
  awsConfig: {
    accessKeyId: AWS_ACCESS_KEY_ID,
    secretAccessKey: AWS_SECRET_ACCESS_KEY,
    region: AWS_REGION,
  },
});

log.info('bla');
```

**Comment by timdp** [comment]
Interesting. I thought log items always had a `level` property. Can you pass your own `formatLog` implementation via the options and output the raw log item that it's trying to format? Thanks!

**Comment by frisibeli** [comment]
I was getting the same error and I managed to fix it by applying this patch:
```
let cwTransport = new CloudWatchTransport(config);
Object.assign(cwTransport, {
      log:(...args) => {
            let callback = args[1];
            cwTransport._relay.submit({date:+new Date(), ...args[0], callback});
      }
})
```

**Comment by timdp** [comment]
@frisibeli Are you on Winston v3 by any chance? It looks like the transport API changed.

**Comment by frisibeli** [comment]
@timdp Yes, my Winston version is winston@3.0.0-rc3

**Comment by timdp** [comment]
That would explain it. I started implementing Winston 3 support on the `winston3` branch today but it's currently untested.

**Comment by bra1n** [comment]
Any idea when this branch is going to be merged into master? As it is, using the branch directly with npm won't work.

**Comment by timdp** [comment]
Thanks for your patience. I've just published v3.0.0 which supports winston 3.
